### DISCUSSION TITLE
styling cells from tags

### DISCUSSION THREAD

**Discussion by parmentelat** [executablebooks/meta#1033] [General]
Trying to style my notebook cells based on the presence of tags
I have the jupyterlab side working fine, and would like to get similar results in the html output of jupyter book
Here's what I can see at this point:

### code cells are fine

if I set a tag, say `level_basic` on one code cell
i.e. its metadata looks like `{'tags' : [ 'level_basic']}`
then in jb's html output the cell is contained in an element like this

```html
<div class="cell tag_level_basic docutils container"> ... </div>
```
so I can style those thanks to the presence of the `tag_level_basic` class

### markdown cells: not so much...

I was naively expecting to observe something similar with markdown cells
Except that it is not the case at all !

Indeed when inspecting the resulting html, it appears that there is no structuring html tag at all that would correspond to the notebook cells; so that for example 2 consecutive markdown cells appear joined in the output, with no remembrance of the initial cell sectioning

and so that's the end of the "styling based on tags" idea altogether, as far as markdown cells at least

---
and so I guess my question is, is this by design ? or is there any chance to amend the tool so that the output reflects the source cells ?

**Comment by chrisjsewell** [comment]
Heya, Notebooks and Markdown documents essentially have incompatible structures:

Notebooks are linearly structured in to cells, e.g. `MD, Code, Md, Raw, ...`

Markdown documents are hierarchically structured by headings into nested sections, e.g. `h1 { h2 { h3, h3 }, h2 { h3 } }`

Code cells are essentially "leaf nodes" within the hierarchical structure, so it's possible to store their styling,
but it is not easy to map the styling of a markdown cell to a particular node(s) in the document tree

**Reply by chrisjsewell** [reply]
(obviously happy to hear potential solutions, but yeh, this issue has always felt a bit intractable to me)
